services/datamanager/builtin/file_deletion_test.go

```diff
@@ -0,0 +1,188 @@
+package builtin
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"os"
+	"sync/atomic"
+	"testing"
+
+	v1 "go.viam.com/api/app/datasync/v1"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/services/datamanager/datasync"
+)
+
+func TestFileDeletionUsageCheck(t *testing.T) {
+	tests := []struct {
+		name                 string
+		expectedFsCheckValue bool
+		testThreshold        float64
```

Comment:
maybe triggerThreshold and captureUsageRatio? I think test is implied by the usage of the test struct below.